<hr>
# การคำนวณหาระยะห่าง
<br>
<hr>

In [88]:
import numpy as np
import matplotlib.pyplot as plt

### ความสูงที่กล้องมองเห็น จากระยะ 20-100 ซม.
* #### ระยะ 20 ซม. เท่ากับ 6 ซม. (front_cm)
* #### ระยะ 100 ซม. เท่ากับ 23.64 ซม. (back_cm)

In [91]:
front_cm = 6
back_cm = 23.64
camera_watch = front_cm / back_cm
camera_watch

0.25380710659898476

### อัตราส่วนของพิเซลต่อความสูงที่กล้องมองเห็น
* #### px_fullsize (ความสูงของพิกเซลที่กล้องมองเห็น)
    * พิกเซลที่กล้องมองเห็นเท่ากับ 240 px

In [92]:
px_fullsize = 240
px_camera = px_fullsize * camera_watch / px_fullsize
px_camera

0.25380710659898476

### ฟังก์ชั่นเข้า scale (ย่อขนาด)
* #### px_point (ค่าพิกเซลที่ตรวจหาได้)

In [93]:
def px_camera_scale(px_point):
    return ((px_point * camera_watch) / px_fullsize) * back_cm
   
px_camera_scale(240)

6.0

### ฟังก์ชั่นแปลงพิกเซลที่ย่อขนาดแล้วให้เป็นเซนติเมตร
* #### px_cm (ค่าพิกเซลที่ย่อแล้ว)

In [94]:
def px_cm(px_cm):
    return px_cm * (back_cm / front_cm) 
px_cm(6)

23.64

### ช่วงของความสูงที่ย่อขนาดแล้ว ที่กล้องสามาถมองเห็นได้
* #### front_cm : ระยะสูงที่กล้องมองเห็นได้ เท่ากับ 6
* #### px_camera : โดยแบ่งเป็นช่วงละ 0.25380710659898476 (1 ซม.)

In [95]:
preriod_cm_carema = np.arange(0, 6, px_camera)
preriod_cm_carema

array([ 0.        ,  0.25380711,  0.50761421,  0.76142132,  1.01522843,
        1.26903553,  1.52284264,  1.77664975,  2.03045685,  2.28426396,
        2.53807107,  2.79187817,  3.04568528,  3.29949239,  3.55329949,
        3.8071066 ,  4.06091371,  4.31472081,  4.56852792,  4.82233503,
        5.07614213,  5.32994924,  5.58375635,  5.83756345])